tests/unit/test_parameters/test_process_parameter_data.py

```diff
@@ -2,56 +2,59 @@
 # Tests for the parameter processing functions
 #
 
-
 import os
 import numpy as np
 import pybamm
-
 import pytest
 
 
+@pytest.fixture
+def path():
+    return os.path.abspath(os.path.dirname(__file__))
+
+
+@pytest.fixture(
+    params=[
+        ("lico2_ocv_example", pybamm.parameters.process_1D_data),
+        ("lico2_diffusivity_Dualfoil1998_2D", pybamm.parameters.process_2D_data),
+        ("data_for_testing_2D", pybamm.parameters.process_2D_data_csv),
+        ("data_for_testing_3D", pybamm.parameters.process_3D_data_csv),
+    ]
+)
+def parameter_data(request, path):
+    name, processing_function = request.param
+    processed = processing_function(name, path)
+    return name, processed
+
```

Comment:
```suggestion
def parameter_data(request, parameters_path):
    name, processing_function = request.param
    processed = processing_function(name, parameters_path)
    return name, processed

```

Reply:
Thanks for reviewing my PR  @agriyakhetarpal !,  I'll implement your suggested changes and update the PR soon.